### ISSUE TITLE
Run cirrus tests periodically

### ISSUE THREAD

**Issue by fulder** [ix-plugin-hub/iocage-plugin-index#194]
I've been trying to make the new cirrus tests more stable and remove any false positives I could think of :crossed_fingers:. But was thinking it could be a good idea to run the cirrus tests (from master) e.g. every night in order to catch bugs such as broken packages or dependency issues before any users installing the plugins will.

What do you think about this idea? 

Created an issue instead of PR as any cirrus cron jobs need to be enabled in the cirrus ui (https://cirrus-ci.com/) inside project settings and cannot be done using the `.cirrus.yml`. See: https://cirrus-ci.org/guide/writing-tasks/#cron-builds

**Comment by tprelog** [comment]
I agree, this would be nice to have. If not nightly, then at least on weekly basis

**Comment by jsegaert** [comment]
I also agree; almost always breakage to the plugins I maintain is caused by some upstream changes in some packages, and not by a PR or change to the plugin itself.

Are those (very cool) CI tests still active?  Somehow my latest PR #222 didn't seem to trigger any tests...

**Comment by fulder** [comment]
@jsegaert the tests should still be active. I'm confused why it didn't run on your PR. Just tried to both create a test MR on my fork with a small change in the same `mineos.json` and it triggered just fine (https://cirrus-ci.com/build/4731071641157632) as well as sync upstream master to one of my currently open PRs here which also triggered the correct task: https://github.com/ix-plugin-hub/iocage-plugin-index/pull/208/checks?check_run_id=2533775947

If I understand the cirrus `changesInclude` function correctly, it should trigger the `mineos` task if the PR has introduced some new changes to the `mineos.json` file (see: https://cirrus-ci.org/guide/writing-tasks/#supported-functions)

Maybe there was some sort of hickup between the CI and GitHub, could you maybe try pushing some dummy commit to your branch and see if it still doesn't work?

**Comment by fulder** [comment]
@kmoore134 and @sonicaj, any thoughts on enabling scheduled tests for this repository in order to catch dependency issues faster? :) (I have a dream (:smile:)  the tests are close to getting all green and once a new dependency is no longer working it would toggle to red status on the `master` branch)

**Comment by jsegaert** [comment]
Also, @kmoore134  and @sonicaj, there are a lot of changes/fixes in `master`, is there any timeline on merging these into `12.2-RELEASE` so the TrueNAS can benefit from our efforts?  😀

**Comment by sonicaj** [comment]
@jsegaert i'll be updating `12.2-RELEASE` as soon as we have https://github.com/ix-plugin-hub/iocage-plugin-index/pull/220 in order :)

**Comment by fulder** [comment]
A short summarize of the current test status after #220 all test run and merge:

Test run: https://cirrus-ci.com/build/4745320211939328

Failing plugins:
* `chia`: Plugin PR fix: https://github.com/kmoore134/iocage-plugin-chia/pull/1
* `motioneye`: This repo PR fix: https://github.com/ix-plugin-hub/iocage-plugin-index/pull/188
* `rtorrent-flood`: Pluing PR fix: https://github.com/pierrickrouxel/iocage-plugin-rtorrent-flood/pull/9
* `sabnzbd`: Issue: https://github.com/ix-plugin-hub/iocage-plugin-index/issues/224, PR fix: https://github.com/ConorBeh/iocage-plugin-sabnzbd/pull/2
